pkg/kubectl/rolebinding.go

```diff
@@ -147,13 +149,20 @@ func (s RoleBindingGeneratorV1) StructuredGenerate() (runtime.Object, error) {
 	}
 	for _, sa := range sets.NewString(s.ServiceAccounts...).List() {
 		tokens := strings.Split(sa, ":")
-		if len(tokens) != 2 {
+		if len(tokens) != 2 || tokens[1] == "" {
 			return nil, fmt.Errorf("serviceaccount must be <namespace>:<name>")
 		}
+		if roleBinding.RoleRef.Kind == "ClusterRole" && tokens[0] == "" {
```

Comment:
For an RBAC rolebinding (even to a cluster role), if the namespace on an SA subject is empty, it uses the local namespace at the authorizer level.  The client doesn't have to set it.  That means that we don't have to fill in a default value.

I know it's a little weird, but the distinction helps when someone later tries to export a rolebinding and re-use it in another namespace.  Ping me on slack more clarification would help.